### PR TITLE
Restrict repository checks to the BCI repo

### PR DESCRIPTION
**Pull request by dirkmueller**
In case the host is registered, it would otherwise use the registered channels

### CODE REVIEW

**Comment by dirkmueller** [comment]
CI is broken due to bsc#1216578

File: bci_tester/data.py [hunk 1]
```diff
@@ -544,17 +551,14 @@ def create_BCI(
 
 REPOCLOSURE_CONTAINER = DerivedContainer(
     base="registry.fedoraproject.org/fedora:latest",
-    containerfile=r"""RUN dnf -y install 'dnf-command(repoclosure)'
+    containerfile=f"""RUN dnf -y install 'dnf-command(repoclosure)'
```
Comment (dcermak) [inline bci_tester/data.py:554]: you can combine `r` and `f`, so no need for the `\` escaping then
```suggestion
    containerfile=rf"""RUN dnf -y install 'dnf-command(repoclosure)'
```

File: bci_tester/data.py [hunk 2]
```diff
@@ -544,17 +551,14 @@ def create_BCI(
...
-    + BCI_DEVEL_REPO
-    + r""" \n\
-priority=100' > /etc/yum.repos.d/SLE_BCI.repo
-""",
+RUN printf '[{BCI_REPO_NAME}]\\n\
+enabled=1\\n\
+name="SLE BCI"\\n\
+autorefresh=0\\n\
+baseurl={BCI_DEVEL_REPO}\\n\
+priority=100\\n' > /etc/yum.repos.d/{BCI_REPO_NAME}.repo""",
```
Comment (dcermak) [inline bci_tester/data.py:561]:
```suggestion
RUN printf '[{BCI_REPO_NAME}] \n\
enabled=1 \n\
name="SLE BCI" \n\
autorefresh=0 \n\
baseurl={BCI_DEVEL_REPO} \n\
priority=100\n' > /etc/yum.repos.d/{BCI_REPO_NAME}.repo""",
```

File: tests/test_repository.py [hunk 3]
```diff
@@ -196,7 +196,10 @@ def test_package_installation(container_per_test, pkg):
     that they are not accidentally not installable.
```
Comment (dcermak) [inline tests/test_repository.py:197]:
```suggestion
    We additionally have to remove the ``container-suseconnect`` zypper service before running the test to ensure that no SLES repositories are added on registered hosts thereby skewing our results.
```

File: tests/test_all.py [hunk 4]
```diff
@@ -148,6 +149,9 @@ def test_lifecycle(auto_container):
                 support_end = datetime.datetime.strptime(
                     entry_date, "%Y-%m-%d"
                 )
+                if entry_name.startswith("go1.19-openssl"):
+                    # bsc#1216578
+                    continue
```
Comment (dcermak) [inline tests/test_all.py:154]: How about something like this? Or is there more lifecycle data to be tested?
```suggestion
                if entry_name.startswith("go1.19-openssl"):
                    pytest.skip(reason="bsc#1216578")
```

Comment (dirkmueller) [reply tests/test_all.py:154]: @dcermak that would abort the loop, we still want the other entries to be tested. I'd have to create a dict with fails and then diff that with the known exceptions. 

Not sure it is worth the effort, but I can do it. I don't plan on adding additional exceptions, I'd rather push people to provide correct data (and get QA to test stuff before releasing it)

File: bci_tester/data.py (hunk 2 above)
Comment (dirkmueller) [reply bci_tester/data.py:561]: whats the point of that additional trailing whitespace? what am I missing?

File: bci_tester/data.py (hunk 1 above)
Comment (dirkmueller) [reply bci_tester/data.py:554]: yeah, don't particularly like raw strings, but I'm not gonna argue about it..

File: tests/test_all.py (hunk 4 above)
Comment (dcermak) [reply tests/test_all.py:154]: then nevermind

File: bci_tester/data.py (hunk 2 above)
Comment (dcermak) [reply bci_tester/data.py]: Just to reduce the diff noise. No point beyond that, so just ignore it.

**Review by dcermak — APPROVED** [review]
LGTM, but the CI needs https://github.com/SUSE/BCI-tests/pull/336

**Pull request merged**